### DISCUSSION TITLE
Infinite loop in i3blocks with Python

### DISCUSSION THREAD

**Discussion by 0x9060** [i3/i3#5668] [Q&A] [answered]
Not sure if this is the right place to ask since it's more about i3blocks...

What I am trying to accomplish is having a blocklet where, upon every interval (e.g., `interval=3600`), an infinite loop script is kicked off. In the beginning of the script, stock data is fetched via API. This part is easy. Thereafter, an infinite loop prints out the i3wm data, full,short,color,bg for each stock, every few seconds. The latter part is actually most difficult...

As a simple example, I cannot even get this to print in my i3bar:
```
#!/usr/bin/env python3

import time

def ticker_tape(quotes):
    full_text = ""
    i = 0
    while True:
        i = (i+1) % len(quotes)
        full_text = f"{quotes[i]}"
        print(f'{full_text}')
        time.sleep(8)


def main():
    ticker_tape(["AAPL", "TSLA"])

if __name__ == "__main__":
    main()
```

This is executable and works normally from my terminal.

Then my i3blocks config is simply:
```
[tickertape]
command=$SCRIPT_DIR/tickertape
interval=persist
```

I've tried with different interval settings [per the docs](https://vivien.github.io/i3blocks/), and none work.

Am I messing something up with the Python script?

**Comment by okraits** [comment, marked as answer]
Some ideas:

1. Verify that the output lines of your script end with a newline (I think they do).
2. Maybe you need to use `print` with `flush=True`.

**Reply by orestisfl** [reply]
Yup, very likely to be that. Also see the FAQ in i3blocks: https://github.com/vivien/i3blocks#faq item 8

**Reply by 0x9060** [reply]
Indeed, that fixed it!

Thanks for the FAQ refernce as well, can't believe I missed that. I thought I've gone entirely through both i3wm user guide + vivien's i3blocks guide about 5x but not quite...
